File: src/Python/weekly_pickup.py
import pandas as pd 
import datetime 
import matplotlib.pyplot as plt 
import numpy as np

all_date = []
#=============================================================================================================#
df = pd.read_csv("uber_nyc_enriched.csv")
df.fillna(value="NA", inplace=True)

for i in enumerate(df["pickup_dt"]):
    all_date.append(datetime.datetime.strptime(i[1], "%Y-%m-%d %H:%M:%S").date().strftime('%A'))

df["pickup_dt"] = all_date

query_table = df.query('pickup_dt == ["Sunday"] and borough == ["Manhattan"]')
sunday = query_table.pickups

query_table = df.query('pickup_dt == ["Monday"] and borough == ["Manhattan"]')
monday = query_table.pickups

query_table = df.query('pickup_dt == ["Tuesday"] and borough == ["Manhattan"]')
tuesday = query_table.pickups

query_table = df.query('pickup_dt == ["Wednesday"] and borough == ["Manhattan"]')
wednesday = query_table.pickups

query_table = df.query('pickup_dt == ["Thursday"] and borough == ["Manhattan"]')
thursday = query_table.pickups

query_table = df.query('pickup_dt == ["Friday"] and borough == ["Manhattan"]')
friday = query_table.pickups

query_table = df.query('pickup_dt == ["Saturday"] and borough == ["Manhattan"]')
saturday = query_table.pickups

data_to_plot = np.array([sunday, monday, tuesday, wednesday, thursday, friday, saturday])

plt.boxplot(data_to_plot)
plt.xlabel('Days')
plt.ylabel('Pickups')
plt.title('Box Plot of Pickups in Manhattan')
plt.xticks(range(1, 8, 1), ('Sun', 'Mon', 'Tue', 'Wed', 'Thurs', 'Fri', 'Sat'))
plt.show()

File: src/R/weekly_pickup.R
library(dplyr)
library(tidyr)
library(ggplot2)

filename <- "uber_nyc_enriched.csv"
df <- read.csv(filename, header = TRUE)

df$pickup_dt <- weekdays(as.POSIXct(df$pickup_dt), abbreviate = FALSE)

query_table = subset(df, df$borough=='Manhattan')
query_table$pickup_dt <- factor(query_table$pickup_dt, c("Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"))

ggplot(query_table, aes(x=pickup_dt, y=pickups))+geom_boxplot()
